source/server/listener_manager_impl.cc

```diff
@@ -1049,6 +1049,15 @@ void ListenerManagerImpl::maybeCloseSocketsForListener(ListenerImpl& listener) {
     // close the socket because they need to receive packets for existing connections via the
     // listen sockets.
     listener.listenSocketFactory().closeAllSockets();
+
+    // In case of this listener was in-place updated previously and in the filter chains draining
+    // procedure, so close the sockets for the previous draining listener.
+    for (auto& manager : draining_filter_chains_manager_) {
+      if (manager.getDrainingListenerTag() == listener.listenerTag()) {
+        manager.getDrainingListener().listenSocketFactory().closeAllSockets();
+        break;
```

Comment:
When the listener config A updates to config B, then updates to config C and both updates go through the in-place update,

A and B may be both in the draining_filter_chains_manager_.

But B is skiped because of the break.

It should be good if the `break` statement is removed

Reply:
good point, thanks! remove the break, also add integration test for this case.